examples/routing_example/server/routes/deprecated_routes/internal.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { IRouter } from '@kbn/core/server';
+import { DEPRECATED_ROUTES } from '../../../common';
+
+export const registerInternalDeprecatedRoute = (router: IRouter) => {
+  router.get(
+    {
+      path: DEPRECATED_ROUTES.INTERNAL_DEPRECATED_ROUTE,
+      validate: false,
+      options: {
+        // Explicitly set access is to internal
+        access: 'internal',
+        deprecated: {
+          documentationUrl: 'https://elastic.co/',
+          severity: 'critical',
+          message: 'Additonal message for internal deprecated api',
+          reason: { type: 'deprecate' },
+        },
```

Comment:
This actually raises something of an oversight: I don't think ever setting an internal route as deprecated was intended for this API 🤔 , although being able to set `critical` on it could be useful if we want to remove it and have known integrations.

My gut feel would be to remove it from the types for internal routes (versioned or non-versioned). FWIW I don't think we should block on it here.

Reply:
Lets address this in a separate PR after we think about it for all API deprecations

Reply:
> My gut feel would be to remove it from the types for internal routes (versioned or non-versioned).
++ internal APIs aren't supposed to be consumed externally and we can add `.jsdocs` to warn plugin developers that the route is deprecated.